modules/javafx.graphics/src/main/java/javafx/scene/Scene.java

```diff
@@ -732,6 +734,21 @@ public final void removePostLayoutPulseListener(Runnable r) {
         postLayoutPulseListeners.remove(r);
     }
 
+    private boolean cleanupAdded = false;
+    private TKPulseListener cleanupListener = () -> {
+        cleanupAdded = false;
+        // JDK-8269907 - This is important, to avoid memoryleaks in dirtyNodes and Parent.removed
+        synchronizeSceneNodesWithLock();
+    };
+    private void checkCleanDirtyNodes() {
+        if(!cleanupAdded) {
+            if((window.get() == null || !window.get().isShowing()) && dirtyNodesSize > 0) {
```

Comment:
Minor: space after `if`.
You could also reduce nesting by consolidating the two separate conditions.